ibm_cloud_sdk_core/authenticators/basic_authenticator.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .authenticator import Authenticator
+from ..utils import has_bad_first_or_last_char
+
+
+class BasicAuthenticator(Authenticator):
+    authentication_type = 'basic'
+
+    def __init__(self, username, password):
+        """
+        :attr str username: The username
+        :attr str password: The password
+        """
+        self.username = username
+        self.password = password
+        self.validate()
+
+    def validate(self):
+        """
+        Performs validation on input params
+        """
+        if self.username is None or self.password is None:
+            raise ValueError('The username and password shouldn\'t be None.')
+
+        if has_bad_first_or_last_char(
+                self.username) or has_bad_first_or_last_char(self.password):
+            raise ValueError(
+                'The username and password shouldn\'t start or end with curly brackets or quotes. '
+                'Please remove any surrounding {, }, or \" characters.')
+
+    def authenticate(self):
+        """
+        Returns the username and password tuple
+        """
+        return (self.username, self.password)
```

Comment:
In the other authenticators the authenticate function returns a string to use for the Authorization header

````python
return 'Bearer {0}'.format(self.bearer_token)
````

Should this return the base 64 encoded u:p? Maybe we can get rid of `_is_bearer_authentication` and `_is_basic_authentication`?

Reply:
> In the other authenticators the authenticate function returns a string to use for the Authorization header

Actually, in Node, I'm passing the request object to the authenticate method and adding the header to it by reference. Maybe that's something that I'll be asked to change but how the authenticate method works could potentially be different from language to language, since it's not user facing. Could be worth discussing.

Reply:
So in python `request` library, the basic auth can be passed directly without encoding using the auth=(username, password) param. And for a bearer token, we have to pass it as a special `Authorization` header.

Reply:
@dpopp07 yeah thats how .NET works as well, by other authenticators i mean the python bearerToken, cp4d and iam authenticators all return strings to add to the header.

Reply:
That makes a lot of sense! My bad on the misunderstanding